ls1intum/Artemis
# Refactor get text submission without assessment

### Checklist

### Motivation and Context


Move code specific to text exercise from `ParticipationResource` to `TextSubmissionResource`.
Fixes #1002 .

### Description

- Now instead of getting participation without assessment for text exercise from `ParticipationResource`, we get submission without assessment from `TextSubmissionResource` in the same manner we do it for file-upload and modeling exercises.
- Completely removed `GET` Submission endpoint, because it was never used by the client. This endpoint didn't hide any details, so tutor can check which student submitted text exercise. This is also the only client change I had to do. This should fix the issue with new assessments not loaded. 
- Additionally to make refactoring more incremental I moved the common logic of two methods `setNewResult` and `lockSubmission` from concrete submission services to common `SubmissionService`.

**Test Coverage**
I added new tests to cover new functionality introduced in `TextSubmissionResource`. The code for `SubmissionService` that was changed is also covered with tests. 
![Screenshot_2020-03-10 de tum in www1 artemis service](https://user-images.githubusercontent.com/17262376/76271318-68ad3200-6278-11ea-9c93-db9568b13559.png)
![Screenshot_2020-03-05 de tum in www1 artemis web rest(3)](https://user-images.githubusercontent.com/17262376/76026163-8acc4a80-5f2e-11ea-81f1-a656bd2cd763.png)
I also update client tests for `TextAssessmentComponent` and it covers almost 85% of code. It covers all logic not related the view manipulation.


### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Select Tutor Dashboard of a course with Text Exercise submissions.
4. Select Exercise Dashboard for Text Exercise
5. Start assessing Text Submission (check that no errors are shown, the submission is correct and result is saved properly).
